app/models/assignment.rb

```diff
@@ -264,15 +264,12 @@ def display_for_note
   end
 
   def total_mark
-    total = 0
-    if self.marking_scheme_type == 'rubric'
-      get_criteria(:ta).each do |criterion|
-        total = total + criterion.weight * 4
-      end
-    else
-      total = get_criteria(:ta).map(&:max).sum()
-    end
-    total.round(2)
+    get_criteria.first.nil? ? 0.round(2) : get_criteria.first.total_mark
+  end
+
+  # Returns the maximum possible mark for a particular assignment
+  def get_max_mark
+    get_criteria.first.nil? ? 0.round(2) : get_criteria.first.get_max_mark
```

Comment:
1. You can probably use `.empty?`
2. Seems like this doesn't add the marks for individual criteria together.